### ISSUE TITLE
opensuse 15.3 fail to install grub2

### ISSUE THREAD

**Issue by cryptearth** [openzfs/openzfs-docs#178]
So, with a few issues here and there in the current list (sorry, I was too lazy to note them all down on the way - but there're quite a few that needs to be fixed) to ultimate end is in step 7 mark 6: 

grub2-install $DISK fails with

grub2-install: error: ../grub-core/kern/fs.c:121:unknown filesystem.

This test was done on an old non-uefi system as well as on a modern uefi system - both fail with the same error at the same step. Current Grub2 version in opensuse 15.3 is 2.04.

**Comment by Zaryob** [comment]
Hi, I wrote the guide about  ZFS on OpenSUSE. When I made the document grub was working. but newer releases it started to not working. Because opensuse doesn't care ZFS due to its license issues. 

And this issue is a known bug: [report](https://bugzilla.opensuse.org/show_bug.cgi?id=1182994)

One of solution is using unstable grub package that provided in [michael-chang's repository](https://download.opensuse.org/repositories/home:/michael-chang:/devel:/grub/openSUSE_Tumbleweed/) But it is complicated to use, if youre not expert.

So, my recommendation is use another distro's grub if you have dualboot with another linux. Or make grub-install using ubuntu iso.

😞

**Comment by cryptearth** [comment]
Thanks for your reply. Well, the grub install error is just one of the issues in the current guide - there're already quite a few other issues before getting to that step which required on-the-fly thinking-out-of-the-box and a few re-runs to at least get there.
As an example: creating rpool/var/mail fails some package to install later on as the rpm install complains the directory already exists. In step 5 there're several issues like thet network config: the auto created links for /etc/hosts, /etc/resolv.conf and a couple others have to be deleted and overridden manual, otherwise after using chroot they break and network doesn't work - the bind mounts using --rbind should be followed by another line using the --make-rslave parameter, or else the unmount later on fails because some of the binds are busy - localectl doesn't work but throwing an error: "system not started by systemd" (asking google there're a few workarounds) - and a couple of others.
Overall the guide is pretty broken right now and even if you get down to install grub it fails due to above reported issue.
As for your recommendation to choose another distro: Not just for ZFS but also for some virtualization stuff I already changed ovver to Ubuntut 21.04 - as it, along with not yet released Debian 11 and recent Fedora 34, are the only three distros (well, there's a 4th vague russian one ...) providing a couple of packages I require for my planed KVM setup (libvirt-daemon-driver-storage-zfs specifically).
As for opensuse itself: I had quite some issues over the years (startitng back in the 12.x or 13.x days) but always managed to get around them - at least on my servers, and at least on them I'll continue to use it. But for my desktop I decided to switch to Ubuntu.
Fun fact: the current Ubuntu 21.04. graphic installer already provide setup on a zfs root right out of the box. So, guess I got two for one on this choice ...
